aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,336 @@
+
+The sample code used in the article is based on the **Localization.StarterWeb** project on `GitHub <https://github.com/aspnet/entropy>`__.
+
+.. contents:: In this article:
+  :local:
+  :depth: 1
+  
+.. note:: Currently, resource files are not read when the project is run from Visual Studio. See `this issue <https://github.com/aspnet/dnx/issues/3047>`_ for more information. Until the issue with Visual Studio is addressed, you can test the project by running it from the command line.
+
+Working with resource files
+-----------------------------
+
+A resource file is a useful mechanism for separating localizable strings from code. Translated strings for the non-default language are isolated *.resx* resource files. For example, you might want to create Spanish resource file named *Welcome.es.resx* containing translated strings. "es" is the language code for Spanish. To create this resource file in Visual Studio:
+
+#. In **Solution Explorer**, right click on the folder which will contain the resource file > **Add** > **New Item**.
+
+.. image:: localization/_static/newi.png
+
+2. In the **Search installed templates** box, enter "resource" and name the file.
+
+.. image:: localization/_static/res.png
+
+3. Enter the key value (native string) in the **Name** column and the translated string in the **Value** column.
+
+.. image:: localization/_static/hola.png
+
+Visual Studio shows the *Welcome.es.resx* file.
+
+.. image:: localization/_static/se.png
+
+Later on in this article I'll show how your ASP.NET app can use the key "Hello" to fetch the string "Hola".
+
+Generating resource files with Visual Studio
+-----------------------------------------------
+
+If you create a resource file in Visual Studio without a culture in the file name (for example, *Welcome.resx*), Visual Studio will create a C# class with a property for each string.  That's usually not what you want with ASP.NET Core; you typically don't have a default *.resx* resource file (A *.resx* file without the culture name). We suggest you create the *.resx* file with a culture name (for example *Welcome.fr.resx*). When you create a *.resx* file with a culture name, Visual Studio will not generate the class file.
+
+If you remove the ".fr" culture designator AND you have the culture set to French (via cookie or other mechanism), the default resource file is read and strings are localized. This is expected behavior. \*.resx designates a default or fallback resource, so that when nothing meets your requested culture you're served \*.resx. If you want to just return the key when missing a resource for the requested culture you must not have a default resource file.
```

Comment:
Maybe this belongs in another section? Seems a bit shoe-horned in here. Anyone else have thoughts?